composition-js/src/merging/merge.ts

```diff
@@ -3120,6 +3131,15 @@ class Merger {
       }
     }
 
+    const linkDirective = this.linkSpec.coreDirective(this.merged);
+    for (const link of linksToPersist) {
+      dest.applyDirective(linkDirective, {
+        url: link.toString(),
+        for: link.defaultCorePurpose,
+        feature: undefined
+      });
+    }
```

Comment:
Two things here:
1. I'm assuming that the `@connect` spec, when used in the supergraph schema, is being used purely as a marker spec.
2. This code will break when multiple versions of the `@connect` spec in subgraphs come about. (In general, there's a better place to put this kind of code.)